<?xml version="1.0" ?>
<sdf version="1.6">
  <model name="camera_pole">
    <link name="base_link">
      <inertial>
        <inertia>
          <ixx>0.0083137104</ixx>
          <iyy>0.0387382402</iyy>
          <izz>0.0469845106</izz>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyz>0</iyz>
        </inertia>
        <pose frame="">0 0 0 0 0 0</pose>
        <mass>1e-10</mass>
      </inertial>
      <collision name="collision">
        <geometry>
          <cylinder>
            <radius>0.02</radius>
            <length>0.4</length>
          </cylinder>
        </geometry>
        <pose frame="">0 0 0.2 0 0 0</pose>
      </collision>
      <visual name="visual_cylinder">
        <geometry>
          <cylinder>
            <radius>0.02</radius>
            <length>0.4</length>
          </cylinder>
        </geometry>
        <pose frame="">0 0 0.2 0 0 0</pose>
      </visual>
      <visual name="visual_sphere">
        <geometry>
          <sphere>
            <radius>0.05</radius>
          </sphere>
        </geometry>
        <pose frame="">0 0 0.4 0 0 0</pose>
      </visual>
    </link>
    <link name="fwd_cam_link">
      <pose frame="">0 0 0.42 0 -1.5708 0</pose>
      <inertial>
        <inertia>
          <ixx>0.1</ixx>
          <iyy>0.1</iyy>
          <izz>0.1</izz>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyz>0</iyz>
        </inertia>
        <pose frame="">0 0 0 0 0 0</pose>
        <mass>1e-10</mass>
      </inertial>
      <sensor name="fwd_cam" type="camera">
        <camera name="fwd_cam">
          <horizontal_fov>2.0943951</horizontal_fov>
          <image>
            <height>720.0</height>
            <width>1280.0</width>
          </image>
          <clip>
            <far>1000.0</far>
            <near>0.1</near>
          </clip>
        </camera>
        <always_on>1</always_on>
        <update_rate>30.0</update_rate>
        <visualize>1</visualize>
        <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <cameraName>fwd_cam</cameraName>
          <imageTopicName>raw</imageTopicName>
          <cameraInfoTopicName>info</cameraInfoTopicName>
          <frameName>fwd_cam_link</frameName>
        </plugin>
      </sensor>
    </link>
    <link name="cam_zero_link">
      <pose frame="">0.02 0 0.4 0 0 0</pose>
      <inertial>
        <inertia>
          <ixx>0.1</ixx>
          <iyy>0.1</iyy>
          <izz>0.1</izz>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyz>0</iyz>
        </inertia>
        <pose frame="">0 0 0 0 0 0</pose>
        <mass>1e-10</mass>
      </inertial>
      <sensor name="cam_zero" type="camera">
        <camera name="cam_zero">
          <horizontal_fov>2.0943951023931953</horizontal_fov>
          <image>
            <height>720.0</height>
            <width>1280.0</width>
          </image>
          <clip>
            <far>1000.0</far>
            <near>0.1</near>
          </clip>
        </camera>
        <always_on>1</always_on>
        <update_rate>30.0</update_rate>
        <visualize>1</visualize>
        <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <cameraName>cam_zero</cameraName>
          <imageTopicName>raw</imageTopicName>
          <cameraInfoTopicName>info</cameraInfoTopicName>
          <frameName>cam_zero_link</frameName>
        </plugin>
      </sensor>
    </link>
    <link name="cam_one_link">
      <pose frame="">1.22465e-18 0.02 0.4 0 0 1.5708</pose>
      <inertial>
        <inertia>
          <ixx>0.1</ixx>
          <iyy>0.1</iyy>
          <izz>0.1</izz>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyz>0</iyz>
        </inertia>
        <pose frame="">0 0 0 0 0 0</pose>
        <mass>1e-10</mass>
      </inertial>
      <sensor name="cam_one" type="camera">
        <camera name="cam_one">
          <horizontal_fov>2.0943951023931953</horizontal_fov>
          <image>
            <height>720.0</height>
            <width>1280.0</width>
          </image>
          <clip>
            <far>1000.0</far>
            <near>0.1</near>
          </clip>
        </camera>
        <always_on>1</always_on>
        <update_rate>30.0</update_rate>
        <visualize>1</visualize>
        <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <cameraName>cam_one</cameraName>
          <imageTopicName>raw</imageTopicName>
          <cameraInfoTopicName>info</cameraInfoTopicName>
          <frameName>cam_one_link</frameName>
        </plugin>
      </sensor>
    </link>
    <link name="cam_two_link">
      <pose frame="">-0.02 2.44929e-18 0.4 0 0 3.14159</pose>
      <inertial>
        <inertia>
          <ixx>0.1</ixx>
          <iyy>0.1</iyy>
          <izz>0.1</izz>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyz>0</iyz>
        </inertia>
        <pose frame="">0 0 0 0 0 0</pose>
        <mass>1e-10</mass>
      </inertial>
      <sensor name="cam_two" type="camera">
        <camera name="cam_two">
          <horizontal_fov>2.0943951023931953</horizontal_fov>
          <image>
            <height>720.0</height>
            <width>1280.0</width>
          </image>
          <clip>
            <far>1000.0</far>
            <near>0.1</near>
          </clip>
        </camera>
        <always_on>1</always_on>
        <update_rate>30.0</update_rate>
        <visualize>1</visualize>
        <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <cameraName>cam_two</cameraName>
          <imageTopicName>raw</imageTopicName>
          <cameraInfoTopicName>info</cameraInfoTopicName>
          <frameName>cam_two_link</frameName>
        </plugin>
      </sensor>
    </link>
    <link name="cam_three_link">
      <pose frame="">-3.67394e-18 -0.02 0.4 0 0 4.71239</pose>
      <inertial>
        <inertia>
          <ixx>0.1</ixx>
          <iyy>0.1</iyy>
          <izz>0.1</izz>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyz>0</iyz>
        </inertia>
        <pose frame="">0 0 0 0 0 0</pose>
        <mass>1e-10</mass>
      </inertial>
      <sensor name="cam_three" type="camera">
        <camera name="cam_three">
          <horizontal_fov>2.0943951023931953</horizontal_fov>
          <image>
            <height>720.0</height>
            <width>1280.0</width>
          </image>
          <clip>
            <far>1000.0</far>
            <near>0.1</near>
          </clip>
        </camera>
        <always_on>1</always_on>
        <update_rate>30.0</update_rate>
        <visualize>1</visualize>
        <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <cameraName>cam_three</cameraName>
          <imageTopicName>raw</imageTopicName>
          <cameraInfoTopicName>info</cameraInfoTopicName>
          <frameName>cam_three_link</frameName>
        </plugin>
      </sensor>
    </link>
    <joint name="fwd_cam_joint" type="fixed">
      <parent>base_link</parent>
      <child>fwd_cam_link</child>
    </joint>
    <joint name="cam_zero_joint" type="fixed">
      <parent>base_link</parent>
      <child>cam_zero_link</child>
    </joint>
    <joint name="cam_one_joint" type="fixed">
      <parent>base_link</parent>
      <child>cam_one_link</child>
    </joint>
    <joint name="cam_two_joint" type="fixed">
      <parent>base_link</parent>
      <child>cam_two_link</child>
    </joint>
    <joint name="cam_three_joint" type="fixed">
      <parent>base_link</parent>
      <child>cam_three_link</child>
    </joint>
  </model>
</sdf>
